book/src/programs.md

```diff
@@ -2,65 +2,67 @@
 
 A client interacts with a Solana cluster by sending it *transactions* with one
 or more *instructions*. The Solana *runtime* passes those instructions to
-user-contributed *programs*. An instruction might, for example, tell a program
-to move *tokens* from one *account* to another or create an interactive
-contract that governs how tokens are moved. Instructions are executed
+user-contributed *services*. An instruction might, for example, tell a
+service to move *tokens* from one *account* to another or create an interactive
+*escrow account* that governs how tokens are moved. Instructions are executed
 atomically. If any instruction is invalid, any changes made within the
 transaction are discarded.
 
-## Deploying Programs to a Cluster
+## Deploying Services to a Cluster
 
 <img alt="SDK tools" src="img/sdk-tools.svg" class="center"/>
 
-As shown in the diagram above a client creates a program and compiles it to an
-ELF shared object containing BPF bytecode and sends it to the Solana cluster.
-The cluster stores the program locally and makes it available to clients via a
-*program ID*. The program ID is a *public key* generated by the client and is
-used to reference the program in subsequent transactions.
+As shown in the diagram above a client creates a service and compiles
+it to an ELF shared object containing BPF bytecode and sends it to the Solana
+cluster. The compiled form is called a *plugin*. The cluster stores the plugin
+locally and makes it available to clients via a *service ID*. The service ID is
+a *public key* generated by the client and is used to reference the service in
+subsequent transactions.
 
-A program may be written in any programming language that can target the
+A service may be written in any programming language that can target the
 Berkley Packet Filter (BPF) safe execution environment. The Solana SDK offers
-the best support for C programs, which is compiled to BPF using the [LLVM
-compiler infrastructure](https://llvm.org). Alternatively, a client might
-choose to bypass LLVM and use Python, Lua or C++ to generate BPF directly via
-the [BPF Compiler Collection](https://github.com/iovisor/bcc) (BCC).
+the best support for services written in C, which is compiled to a plugin using
```

Comment:
`s/plugin/shared object/g`?

Reply:
Hmm, actually why do we even care to distinguish between "service" and "plugin"/"shared object".   We're splitting split hairs at this point.